### ISSUE TITLE
Flow type checker documentation

### ISSUE THREAD

**Issue by avoid3d** [avoid3d/waverbase#1]
I think we should use flow (http://flowtype.org/).

TODO:

- [ ] Document what flow is, why we use in the wiki.
- [ ] Add installation instructions and a link to the wiki to the Readme.

**Comment by lzbotha** [comment]
Flow seems interesting. My javascript is not great but the static type checking seems like it could save you a bunch of headaches. The ability to mix and match typing and no typing worries me a little bit. I fear the untyped code will spread until it encompasses everything.

**Comment by avoid3d** [comment]
@lzbotha yeah the 'boundary' between typed and untyped is often a bit messy, but we have a rule which dictates that all of our own code is typed, which runs before you commit :)

**Comment by lzbotha** [comment]
That seems reasonable. What are we going to be using js for? The site which sells our service?
